model_analyzer/triton/server/server_docker.py

```diff
@@ -180,7 +177,17 @@ def stop(self):
             self._tritonserver_container.stop()
             self._tritonserver_container.remove(force=True)
             self._tritonserver_container = None
-            self._docker_client.close()
+            logger.info('Stopped Triton Server.')
+        else:
+            try:
+                # Possible that an error left a stopped triton container
+                self._tritonserver_container = self._docker_client.containers.get(
```

Comment:
Isn't this going to mistakenly remove other containers named `tritonserver`? I believe we do not set the name of the container to be `tritonserver`. It is automatically chosen using the Docker API.

Reply:
We set the name of our triton server container to be `tritonserver`. The goal of this is to remove any containers named `tritonserver`. We've been running into some issues where if there is an error launching the container, there is a stopped container named `tritonserver` that prevents other tests from running on that node.

Reply:
I see. I was wrong about not setting the container name. But still if the user has another container named `tritonserver` (unrelated to MA) and tries to run the MA, is this going to remove that container?

Reply:
Yes, perhaps the exact name should be changed? Or maybe We should ignore this problem altogether and not name our container? The downside to that approach is that we won't know if we're improperly clearning up stopped containers.

Reply:
Let me try removing the name for now. With the labels we are adding to the CI tests (RUNNER_ID=$RUNNER_ID), the cleanup daemon will remove our stopped containers.

Reply:
I think we used to not name the container.
> The downside to that approach is that we won't know if we're improperly clearning up stopped containers. 

I think if we are able to find the root cause for when the dangling container is created, we wouldn't need this try block. Alternatively, we can extract the name of the container assigned by docker and delete it here.

Reply:
The root cause I've found is what this try block aims to handle. So in certain cases (such as if the image is empty), the docker API throws an exception. In this case, it seems the container is started, exits, and is stopped. Now we don't have a handle (`self._tritonserver_container` is `None`), and there is a stopped container named `tritonserver`. 

We can add checks that prevent these types of scenarios from occuring. In this PR I added a null check for the image, but this try/block is to handle the case where something goes wrong. 

The only reason to keep a named container is to expose cases where we are hanging containers it seems like.